xds/internal/balancer/edsbalancer/edsbalancer_test.go

```diff
@@ -409,25 +409,35 @@ func TestEDS_EndpointsHealth(t *testing.T) {
 	})
 	edsb.HandleEDSResponse(clab1.build())
 
-	var readySCs []balancer.SubConn
+	var (
+		readySCs []balancer.SubConn
+
+		wantNewSubConnAddrStrs = []string{
+			testEndpointAddrs[0],
+			testEndpointAddrs[2],
+			testEndpointAddrs[6],
+			testEndpointAddrs[8],
+		}
+	)
 	for i := 0; i < 4; i++ {
+		addr := <-cc.newSubConnAddrsCh
+		if addr[0].Addr != wantNewSubConnAddrStrs[i] {
```

Comment:
Will they always happen in order?

Optional assuming they _are_ always ordered for now: would a `map[string]bool` work better for `wantNewSubConnAddrStrs`?  It could be more future-proof in case the order changes later.  To ensure it doesn't do the same address four times, you could `delete` the entries from the map as they are pulled in from the channel.

Reply:
Now with sort and cmp.Equal